nsxt/resource_nsxt_edge_transport_node.go

```diff
@@ -2007,12 +2055,23 @@ func setIPv6AssignmentInSchema(spec *data.StructValue) (interface{}, error) {
 	return []interface{}{elem}, nil
 }
 
-func setHostSwitchProfileIDsInSchema(hspIDs []mpmodel.HostSwitchProfileTypeIdEntry) []interface{} {
+func setHostSwitchProfileIDsInSchema(hspIDs []mpmodel.HostSwitchProfileTypeIdEntry, parentMap map[string]interface{}) {
 	var hostSwitchProfileIDs []interface{}
 	for _, hspID := range hspIDs {
+		if hspID.Key == nil {
+			continue
+		}
+		if *hspID.Key == mpmodel.BaseHostSwitchProfile_RESOURCE_TYPE_UPLINKHOSTSWITCHPROFILE {
+			parentMap["uplink_profile"] = hspID.Value
+		}
+		if *hspID.Key == mpmodel.BaseHostSwitchProfile_RESOURCE_TYPE_VTEPHAHOSTSWITCHPROFILE {
+			parentMap["vtep_ha_profile"] = hspID.Value
+		}
 		hostSwitchProfileIDs = append(hostSwitchProfileIDs, hspID.Value)
 	}
-	return hostSwitchProfileIDs
+
+	// deprecated way of specifying profiles in single list
+	parentMap["host_switch_profile"] = hostSwitchProfileIDs
```

Comment:
With this logic, the same profile is being added both to host_switch_profile and to the profile_specific_attributes.

This seems safe, but also made me think that we will keep adding the auto-assigned uplink profile. Why this does not cause a permadiff?

Reply:
`host_switch_profile` is Computed, so if the user switched to the new way of configuring profiles, there will be no permadiff. 
If the user keeps using `host_switch_profile` for configuring one profile type, while the other type is being auto-assigned, the permadiff will occur. Thats why we're deprecating `host_switch_profile` and recommending to use the new attributes.